R/jagsModule.R

```diff
@@ -112,39 +112,52 @@ JAGS <- function(jaspResults, dataset, options, state = NULL) {
       inits[[i]]$.RNG.seed <- runif(1, 0, 2^31)
     }
   }
-  
+
   # this code is similar to how R2jags does it, but with
   # a try around it.
   e <- try({
-    
-    # compile model
-    model <- rjags::jags.model(
-      file     = modelFile,
-      n.chains = noChains,
-      n.adapt  = 0L,
-      data     = datList,
-      inits    = inits#unname(lapply(inits, list))
-    )
-    
-    # sample burnin
-    rjags::adapt(
-      object         = model,
-      n.iter         = noBurnin,
-      by             = 0L,
-      progress.bar   = "none",
-      end.adaptation = TRUE
-    )
-    
-    # sample remainder
-    samples <- rjags::coda.samples(
-      model          = model,
-      variable.names = parametersToSave,
-      n.iter         = noSamples,
-      thin           = noThinning,
-      by             = 0L,
-      progress.bar   = "none"
-    )
-    
+
+    if (isTRUE(options$runUnitTests)) {
```

Comment:
I'd prefer to keep the unit testing stuff within the test files.
What do you think about
```
old.jags.model <- rjags::jags.model
new.jags.model <- function(...) return(NULL)

old.adapt <- rjags::adapt
new.adapt <- function(...) return(NULL)

old.coda.samples <- rjags::coda.samples
new.coda.samples <- function(...) {
  candidates <- c("jags-test-model.rds", file.path("tests", "testthat", "jags-test-model.rds"))
  testFile <- Filter(file.exists, candidates)
  return(readRDS(testFile))
}

jaspTools:::replaceFn("jags.model",   new.jags.model,   "rjags")
jaspTools:::replaceFn("adapt",        new.adapt,        "rjags")
jaspTools:::replaceFn("coda.samples", new.coda.samples, "rjags")

on.exit({
  jaspTools:::replaceFn("jags.model",   old.jags.model,   "rjags")
  jaspTools:::replaceFn("adapt",        old.adapt,        "rjags")
  jaspTools:::replaceFn("coda.samples", old.coda.samples, "rjags")
})
```

Then we don't need to touch the source

Reply:
I guess you could also decide to just overwrite the `coda.samples` function

Reply:
I decided to only overwrite `coda.samples`, then at least the functionality of the other 3 can be checked (e.g., when a model doesn't compile then `jasg.model` crashes).